tools/update_runnablejobs.py

```diff
@@ -1,98 +1,201 @@
+        task_id = latest_task['taskId']
     except Exception as error:
         # we will end this function if got exception here
-        logger.debug("the request to %s failed, due to %s" % (url, error))
-        return
-    task_id = latest_task['taskId']
+        LOG.exception("The request for %s failed due to %s" % (url, error))
+        return None
 
-    # The format of expires is like 2017-07-04T22:13:23.248Z and we only want 2017-07-04 part
-    expires = latest_task['expires'].split('T')[0]
-    time_tuple = datetime.datetime.strptime(expires, "%Y-%m-%d").timetuple()
-    new_timestamp = time.mktime(time_tuple)
-    path = get_rootdir() + '/runnablejobs.json'
+    path = os.path.join(get_rootdir(), '%s.json' % task_id)
 
     # we do nothing if the timestamp of runablejobs.json is equal with the latest task
     # otherwise we download and update it
     if os.path.isfile(path):
-        with open(path, 'r') as data:
-            # read the timesstamp of this task from json file
-            oldtime = json.loads(data.read())['meta']['timetamp']
-        if oldtime == new_timestamp:
-            print "The runnable json file is latest already."
-            return
-        else:
-            print "It's going to update your runnable jobs data."
-            data = query_the_runnablejobs(new_timestamp, task_id)
-            new_jobs = add_jobs_to_jobpriority(new_data=data,
-                                               priority=1,
-                                               timeout=0,
-                                               set_expired=True)
-            with open(path, 'w') as f:
-                json.dump(data, f)
+        LOG.info("We have already processed the data from this task (%s)." % task_id)
+        return None
     else:
-        print "It's going to help you download the runnable jobs file."
-        data = query_the_runnablejobs(new_timestamp, task_id)
-        if data:
-            with open(path, 'w') as f:
-                json.dump(data, f)
-            add_jobs_to_jobpriority(new_data=data, priority=5, timeout=5400)
-
-
-def query_the_runnablejobs(new_timestamp, task_id=None):
-    if task_id:
-        url = TREEHERDER_HOST.format('mozilla-inbound', task_id)
-        try:
-            data = retry(requests.get, args=(url, ), kwargs={'headers': headers}).json()
-            if len(data['results']) > 0:
-                data['meta'].update({'timetamp': new_timestamp})
-        except:
-            print "failed to get runnablejobs via %s" % url
-            print "We have something wrong with the runnable api, switch to the old runnablejobs.json now"
-            with open(get_rootdir() + '/runnablejobs.json', 'r') as data:
-                data = json.loads(data.read())
+        LOG.info("We're going to fetch new runnable jobs data.")
+        # We never store the output of runnable api but the minimal data we need
+        data = sanitized_data(query_the_runnablejobs(repo_name=repo_name, task_id=task_id))
```

Comment:
I am still worried about the case where we end up with an exception whilst calling sanitized_data(), it would be nice to know that we can always fall back to old data.